### DISCUSSION TITLE
Your environment must have WebSockets enabled

### DISCUSSION THREAD

**Discussion by ttrakker** [coder/code-server#3885] [Q&A]
Hi I am trying to setup my own server to learn python with and trying to setup a centos8 box to use. One of the requirements on https://coder.com/docs/code-server/v3.11.0/requirements says "Your environment must have WebSockets enabled, since code-server uses WebSockets for communication between the browser and the server.". Can someone help me understand what that means and how i do so? Googling im confused with every result. thanks

**Comment by jsjoeio** [comment]
Hey there! So WebSockets is a protocol, similar to HTTP(S). 

You're using a CentOS box. How do you plan to install and run code-server? Directly on that box I assume? 

Depending on where it's hosted, you may need to enable WebSockets in order for everything to work properly. Where are you hosting your box?

**Comment by ttrakker** [comment]
Hi thanks for your reply. I just want to host on my home network on an esxi
host.

Not worried about remote access outside my home.

Thanks

**Reply by jsjoeio** [reply]
> Hi thanks for your reply. I just want to host on my home network on an esxi
host.

I'm not familiar with an esxi host, but I assume this happens because you're accessing via your home network. Have you tried following [this section](https://coder.com/docs/code-server/v3.11.1/guide#https-and-self-signed-certificates) of the docs?
